sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -64,6 +64,13 @@ __ESIMD_API SurfaceIndex get_surface_index(AccessorTy acc) {
                 sycl::detail::acc_properties::is_local_accessor_v<AccessorTy>) {
     return detail::SLM_BTI;
   } else {
+#ifdef __ESIMD_FORCE_STATELESS_MEM
+    static_assert(sycl::detail::acc_properties::is_image_accessor_v<AccessorTy>,
+                  "The function get_surface_index() is available only for "
```

Comment:
kind of similar question, i didn't expect we would want to call get_surface_index in stateless mode for anything, even for image and local accessors.